docs/java/features/resilience/caching.mdx

```diff
@@ -0,0 +1,189 @@
+
+```java
+// Set a parameter value
+cacheConfig = CacheConfiguration
+        .of(Duration.ofDays(1))
+        .withParameters("value1");
+resilienceConfig.cacheConfiguration(cacheConfig);
+
+// Cache is empty, so this will be a cache miss
+ResilienceDecorator.executeSupplier(() -> operation(), resilienceConfig);
+// Cache is hit, cached value will be returned
+ResilienceDecorator.executeSupplier(() -> operation(), resilienceConfig);
+
+// Set a different parameter value
+cacheConfig = CacheConfiguration
+        .of(Duration.ofDays(1))
+        .withParameters("value2");
+resilienceConfig.cacheConfiguration(cacheConfig);
+
+// Cache is not hit because the existing cache entry has a different parameter value
+ResilienceDecorator.executeSupplier(() -> operation(), resilienceConfig);
+
+cacheConfig = CacheConfiguration
+        .of(Duration.ofDays(1))
+        .withParameters("value1");
+resilienceConfig.cacheConfiguration(cacheConfig);
+
+// The cached value from the very first invocation is returned as the parameter is "value1" again
+ResilienceDecorator.executeSupplier(() -> operation(), resilienceConfig);
+```
+
+So you can use parameters to separate cache entries as you see fit.
+Please note that there is no need to do this to achieve multi tenancy.
+The SDK is tenant aware by default as described below.
+
+### Multi Tenancy
+
+By default this caching functionality is tenant aware.
+If a tenant is available the SDK will create cache entries only for that specific tenant.
+Other tenants will not see the value cached for that specific tenant.
+
+For this example assume that `config` is a `ResilienceConfiguration` that has a `CacheConfiguration` in place:
+
+```java
+cachedOperation = ResilienceDecorator.decorateSupplier(() -> operation(), config)
+
+// cachedOperation() will be evaluated
+TenantAccessor.executeWithTenant(tenantA, () -> cachedOperation());
+// cached value will be taken
+TenantAccessor.executeWithTenant(tenantA, () -> cachedOperation());
+
+// cachedOperation() will be evaluated because the cached value is only accessible for tenantA
+TenantAccessor.executeWithTenant(tenantB, () -> cachedOperation());
+// cached value for tenantB will be taken
+TenantAccessor.executeWithTenant(tenantB, () -> cachedOperation());
+```
+
+This isolation is configured via the isolation mode of the resilience configuration.
+So you can disable tenant aware caching by setting the isolation mode to `NO_ISOLATION`.
+See the section on isolation levels under [Resilience](https://sap.github.io/cloud-sdk/docs/java/features/resilience/resilience#multi-tenancy)
+
+## Manipulating the State of Caches
+
+:::note
+A specific cache instance is not yet created when configuration is created.
+Instead it is only created when an operation is decorated or executed by the  `ResilienceDecorator`.
+:::
+
+To invalidate the cache you can use:
+```java
+// invalidate
+ResilienceDecorator.invalidateCache(ResilienceConfiguration.of(resilienceId));
```

Comment:
```suggestion
ResilienceDecorator.invalidateCache(resilienceConfig);
```

We have used this object above in the documentation so I think we can use it here as well. Otherwise the user might think he has to re-create the config to do cache invalidation.